internal/querycoordv2/meta/target_manager.go

```diff
@@ -3,184 +3,278 @@ package meta
 import (
 	"sync"
 
+	"github.com/samber/lo"
+	"go.uber.org/zap"
+
 	"github.com/milvus-io/milvus/internal/log"
 	"github.com/milvus-io/milvus/internal/proto/datapb"
-	"github.com/milvus-io/milvus/internal/util/funcutil"
-	"go.uber.org/zap"
+	"github.com/milvus-io/milvus/internal/util/typeutil"
+)
+
+type TargetScope = int
```

Comment:
int32(or any other type with fixed size) is better, which can be applied with atomic operations